internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3187,6 +3223,27 @@ func expandKubernetesClusterWorkloadAutoscalerProfile(input []interface{}, d *pl
 	return &workloadAutoscalerProfile
 }
 
+func expandKubernetesClusterNodeProvisioningProfile(input []interface{}) *managedclusters.ManagedClusterNodeProvisioningProfile {
+	if len(input) == 0 {
+		return nil
+	}
+	config := input[0].(map[string]interface{})
```

Comment:
There is a potential panic here since all attributes are optional and it can be an array of one `nil` instead of one empty map.